nodes/parent/pulp_node/distributors/http/distributor.py

```diff
@@ -214,8 +215,11 @@ def _add_repository(self, repo_id, payload):
         :param payload: The repository payload
         :type payload: dict
         """
-        manager = factory.repo_query_manager()
-        payload['repository'] = manager.get_repository(repo_id)
+        repo_obj = model.Repository.objects.get_repo_or_missing_resource(repo_id)
+        # Pseudo serialize the repository object so that it can be used by a node.
+        payload['repository'] = {'id': repo_obj.repo_id, 'display_name': repo_obj.display_name,
```

Comment:
The scratchpad is necessary so the repository serializer won't work as is. Also, @jortel and I didn't like the idea of using a view serializer to change what is being sent to pulp_nodes.